pkg/reconciler/finalizer/reposync_finalizer_test.go

```diff
@@ -271,11 +279,176 @@ func TestRepoSyncFinalize(t *testing.T) {
 			expectedStopped:            true,
 			expectedRsyncAfterFinalize: nil,
 		},
+		{
+			name:           "orphan happy path",
+			rsync:          repoSync1.DeepCopy(),
+			deletionPolicy: metadata.DeletionPropagationPolicyOrphan,
+			expectedRsyncBeforeDestroy: func() client.Object {
+				obj := repoSync1.DeepCopy()
+				// +1 to set ReconcilerFinalizing condition
+				obj.SetResourceVersion("2")
+				obj.Status.Conditions = []v1beta1.RepoSyncCondition{
+					{
+						Type:    v1beta1.RepoSyncReconcilerFinalizing,
+						Status:  metav1.ConditionTrue,
+						Reason:  "ResourcesDeleting",
+						Message: "Deleting managed resource objects",
+					},
+				}
+				return obj
+			}(),
+			expectedError:   nil,
+			expectedStopped: true,
+			expectedRsyncAfterFinalize: func() client.Object {
+				obj := repoSync1.DeepCopy()
+				// +1 to set ReconcilerFinalizing condition
+				// +1 to remove ReconcilerFinalizing condition
+				// +1 to remove Finalizer
+				// TODO: optimize by combining consecutive updates
+				obj.SetResourceVersion("4")
+				// Finalizer has been removed
+				obj.SetFinalizers(nil)
+				// ReconcilerFinalizing condition added and then removed
+				return obj
+			}(),
+		},
+		{
+			name:           "orphan recovery",
+			deletionPolicy: metadata.DeletionPropagationPolicyOrphan,
+			rsync: func() client.Object {
+				obj := repoSync1.DeepCopy()
+				// +1 to set ReconcilerFinalizing condition
+				// +1 to set ReconcilerFinalizerFailure condition
+				obj.SetResourceVersion("3")
+				// Finalizer NOT removed
+				// ReconcilerFinalizing condition added and NOT removed
+				// ReconcilerFinalizerFailure condition added
+				obj.Status.Conditions = []v1beta1.RepoSyncCondition{
+					{
+						Type:    v1beta1.RepoSyncReconcilerFinalizing,
+						Status:  metav1.ConditionTrue,
+						Reason:  "ResourcesDeleting",
+						Message: "Deleting managed resource objects",
+					},
+					{
+						Type:    v1beta1.RepoSyncReconcilerFinalizerFailure,
+						Status:  metav1.ConditionTrue,
+						Reason:  "DestroyFailure",
+						Message: "Failed to delete managed resource objects",
+						Errors: []v1beta1.ConfigSyncError{
+							{
+								Code:         "2002",
+								ErrorMessage: "KNV2002: example message: APIServer error: destroy error\n\nFor more information, see https://g.co/cloud/acm-errors#knv2002",
+							},
+						},
+					},
+				}
+				return obj
+			}(),
+			expectedRsyncBeforeDestroy: func() client.Object {
+				obj := repoSync1.DeepCopy()
+				// No changes - continue deleting
+				// TODO: Should finalizer re-entry be vissible to the user by toggling a condition?
```

Comment:
not sure what this is asking

Reply:
it's just copying existing boilerplate. I didn't really follow the TODO but I can remove if you prefer

Reply:
Hmm. I guess the question was whether the condition should flip back and forth when attempting and retrying. Based on more recent work in the RGC, I think it's undesirable for conditions to flap back and forth. So the answer is probably NO.

So it should be safe to remove these TODOs.